### PR TITLE
Update test verify that normal user is able to submit jobs

### PR DESCRIPTION
**Pull request by abhijeet-dhumal**
# Issue link

[RHOAIENG-7221](https://issues.redhat.com/browse/RHOAIENG-7221)

# What changes have been made

Execution flow :-
- In test-namepace, create Kueue resources(ClusterQueue and LocalQueue) as an admin user
- Create Configmap to store demo-notebook (mnist_ray_mini.ipynb), training script(mnist.py) and requirements.txt
- Use non-admin user name and login bearer token to create role-binding to allow namespace specific admin access to a non-admin user 
-  Create notebook using related configuration by creating custom resource to run/interact with notebook file using [Papermill](https://papermill.readthedocs.io/en/latest/) tool (mounted using PVC which consists of cofigmap). (This will run demo-notebook in jupyter notebook-pod and will assert Job completion status at the end of Jupyter-notebook pod logs)
-  As part of demo-notebook execution, ray-cluster provisioning, job submission and Job status assert will be executed. 

# Verification steps

1.  Export variables : 
```
export NOTEBOOK_USER_NAME=<normal-user-name> \
NOTEBOOK_USER_TOKEN=<normal-user-bearer-token> \
ODH_NAMESPACE=redhat-ods-applications \
NOTEBOOK_IMAGE_STREAM_NAME=habana-notebook
```
2. Run test :  `go test -v ./test/odh -run TestMnistRay`

## Checks
- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: test/odh/notebook.go [hunk 1]
```diff
@@ -36,44 +37,43 @@ const recommendedTagAnnotation = "opendatahub.io/workbench-image-recommended"
...
-func createNotebook(test Test, namespace *corev1.Namespace, notebookAdminToken, notebookUserToken, localQueue, jupyterNotebookConfigMapName, jupyterNotebookConfigMapFileName string) {
+func createNotebook(test Test, namespace *corev1.Namespace, notebookUserToken, localQueue, jupyterNotebookConfigMapName, jupyterNotebookConfigMapFileName string) {
 	// Create PVC for Notebook
 	notebookPVC := CreatePersistentVolumeClaim(test, namespace.Name, "10Gi", corev1.ReadWriteOnce)
 
 	// Retrieve ImageStream tag for
 	is := GetImageStream(test, GetOpenDataHubNamespace(test), GetNotebookImageStreamName(test))
 	recommendedTagName := getRecommendedImageStreamTag(test, is)
 
+	fmt.Println("LocalQueue Prop : ", localQueue)
```
Comment (sutaakar) [inline test/odh/notebook.go:62]: I think this isn't needed

File: test/odh/mnist_ray_test.go [hunk 2]
```diff
@@ -118,8 +72,48 @@ func TestMnistRay(t *testing.T) {
...
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	_, err := test.Client().Core().RbacV1().RoleBindings(namespace.Name).Create(context.TODO(), roleBinding, metav1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating RoleBinding: %v", err)
+	} else {
+		fmt.Printf("RoleBinding created successfully for user %s with admin access in namespace %s\n", userName, namespace.Name)
+	}
```
Comment (sutaakar) [inline test/odh/mnist_ray_test.go:113]: This can be moved to https://github.com/project-codeflare/codeflare-common/blob/main/support/rbac.go

Comment (abhijeet-dhumal) [reply test/odh/mnist_ray_test.go]: Yes, i have migrated it to https://github.com/project-codeflare/codeflare-common/pull/50
Thank you @sutaakar for reviewing!

File: test/odh/mnist_ray_test.go [hunk 3]
```diff
@@ -118,8 +69,24 @@ func TestMnistRay(t *testing.T) {
...
+	config := CreateConfigMap(test, namespace.Name, map[string][]byte{
+		// MNIST Ray Notebook
+		jupyterNotebookConfigMapFileName: ReadFile(test, "resources/mnist_ray_mini.ipynb"),
+		"mnist.py":                       readMnistPy(test),
+		"requirements.txt":               readRequirementsTxt(test),
+	})
+
+	// Define the regular(non-admin) user
+	userName := GetNotebookUserName(test)
+	userToken := GetNotebookUserToken(test)
```
Comment (sutaakar) [inline test/odh/mnist_ray_test.go:83]: Can you also remove `GetNotebookAdminName` and `GetNotebookAdminToken` functions?
They are not needed any more.

File: go.sum [hunk 4]
```diff
@@ -358,10 +358,12 @@ github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
 github.com/pkg/errors v0.9.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
 github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
-github.com/project-codeflare/appwrapper v0.7.0 h1:hlsc7xI3rHLhtVlspxb0qHkPoAggdWyRUQkRdMOT20A=
-github.com/project-codeflare/appwrapper v0.7.0/go.mod h1:f4SYPB6sPteegtekwD4bjPdpWZ+nFlLF2Eoz9iyw73s=
+github.com/project-codeflare/appwrapper v0.8.0 h1:vWHNtXUtHutN2EzYb6rryLdESnb8iDXsCokXOuNYXvg=
+github.com/project-codeflare/appwrapper v0.8.0/go.mod h1:FMQ2lI3fz6LakUVXgN1FTdpsc3BBkNIZZgtMmM9J5UM=
 github.com/project-codeflare/codeflare-common v0.0.0-20240415144758-8ab7f860e594 h1:p7U6KndosCbA7Rbes5YnXDBFKMj7IsEQRoz39LePZRQ=
 github.com/project-codeflare/codeflare-common v0.0.0-20240415144758-8ab7f860e594/go.mod h1:LoZMDO3zgkl0OoPz0bDVI4WFC3wQ8xE23J2ncsvjy+g=
```
Comment (sutaakar) [inline go.sum:364]: Please run `go mod tidy`

**Review by sutaakar — APPROVED** [review]

**Pull request merged**